Sources/NIOHTTP1/HTTPServerPipelineHandler.swift

```diff
@@ -414,6 +447,28 @@ public final class HTTPServerPipelineHandler: ChannelDuplexHandler, RemovableCha
         context.fireChannelInactive()
     }
 
+    public func close(context: ChannelHandlerContext, mode: CloseMode, promise: EventLoopPromise<Void>?) {
+        var shouldRead = false
+
+        if mode == .output {
+            // We need to do special handling here. If the server is closing output they don't intend to write anymore.
+            // That means we want to drop anything up to the end of the in-flight request.
+            self.dropAllButInFlightRequest()
+            self.state.closeOutputSent()
+
+            // If there's a read pending, we should deliver it after we forward the close on.
+            shouldRead = self.readPending
+        }
+
+        context.close(mode: mode, promise: promise)
+
+        // Double-check readPending here in case something weird happened.
+        if shouldRead && self.readPending {
+            self.readPending = false
+            context.read()
```

Comment:
Mind adding a comment here that because of the state transition to `closeOutputSent()` we will _not_ actually deliver any reads to any handlers further back? We're just dropping them on the floor, that's a good idea but I first didn't get that :)

Reply:
Sure, I've just done that.